tests/analytical.py

```diff
@@ -1,88 +1,113 @@
 import numpy as np
 
 
-def gradient(x_a, y_a, z_a, x_b, y_b, z_b, c_a, c_b):
+def gradient(coordinates,charges):
```

Comment:
I don't think the expression for the gradient you are implementing is entirely correct. If you only have two atoms, then:
![dffa0427077639c231eaefce54e63473](https://user-images.githubusercontent.com/3708689/100465107-07134900-30cf-11eb-9c50-2d862e818e39.png)
and the gradient with respect to atom 0 (for example) is:
![89f21e761da3a8768b7a67ebfc519333](https://user-images.githubusercontent.com/3708689/100465250-417ce600-30cf-11eb-8bf1-bdb6685ce903.png)

Now, if we have 3 atoms:
![9cc49e84e878a8f3e26ef42a0db5ea47](https://user-images.githubusercontent.com/3708689/100465442-99b3e800-30cf-11eb-98e7-f110fed67b66.png)
and the gradient with respect to atom 0:
![04bded84ac9d28abd07d804de4776260](https://user-images.githubusercontent.com/3708689/100465511-bea85b00-30cf-11eb-963a-f938ef111cb8.png)

So this line:
```python
d_ax = -(coordinates[atom_a,0] - coordinates[atom_b,0]) * cxc / xyz_squared
```
should really be:
```python
d_ax -= (coordinates[atom_a,0] - coordinates[atom_b,0]) * cxc / xyz_squared
```
because you're accumulating the result of differentiating with respect to all atoms other than atom 0.